File: javascript/bob/bob.js

var Bob = function() {};

Bob.prototype.hey = function(sentence) {
    if (sentence.match(/[A-Z]{4,}/g)) {
        return 'Whoa, chill out!'
    } else if (sentence.match(/[A-Z]{2,}[!]$/g)) {
        return 'Whoa, chill out!'
    } else if (sentence.match(/[?]$/g)) {
        return 'Sure.'
    } else if (sentence.match(/^(?![\s\S])/g)) {
        return 'Fine. Be that way!'
    } else if (sentence.match(/^\s/g)) {
        return 'Fine. Be that way!'
    } else {
        return 'Whatever.'
    }

};

module.exports = Bob;

File: go/bob/bob.go
package bob // package name must match the package name in bob_test.go
import (
	"strings"
)

const testVersion = 2 // same as targetTestVersion

func Hey(input string) string {

	if input == "" {
		return "Fine, be that way"
	}

	if question(input) {
		return "Sure"
	}

	if yelling(input) {
		return "Whoa, chill out!"
	}

	return "Whatever."
}

func question (sentence string) bool {
	return strings.Contains(sentence, "?")

}

func yelling (sentence string) bool {
	if strings.Contains(sentence, "!") {
		return true
	}

}